query/call_iterator.go

```diff
@@ -53,6 +53,8 @@ func NewCallIterator(input Iterator, opt IteratorOptions) (Iterator, error) {
 		return newLastIterator(input, opt)
 	case "mean":
 		return newMeanIterator(input, opt)
+	case "time_weighted_average":
```

Comment:
This section is meant for functions that can be pushed down to the storage engine so anything that is in this section will effectively be called twice. All of the implementations here are idempotent except for `count` which has a special implementation so that count is performant.

For a function to be in this section, this must be true:

```
X(X(y)) = X(y)
```

Count is special cased so that it does this:

```
sum(count(y)) = count(y)
```

Here's the place where that happens: https://github.com/influxdata/influxdb/blob/e9bada090f10ea04f07ab56a839690ab15328269/query/iterator.go#L139

It is much safer to add this to the appropriate place in `query/select.go`.

Reply:
Thanks for a tip, i will be happy to move it into buildCallIterator in query/select.go. Although there are 2 separate places where switch expr.Name takes place and I am not sure why is that the case and which section time_weighted_average belongs to